src/osd/PrimaryLogPG.cc

```diff
@@ -13631,6 +13654,15 @@ int PrimaryLogPG::rep_repair_primary_object(const hobject_t& soid, OpRequestRef
 	   << " peers osd.{" << actingbackfill << "}" << dendl;
 
   if (!is_clean()) {
+    if (is_degraded_or_backfilling_object(soid)) {
```

Comment:
I'm not clear that this is a permanent condition, that requires an EIO.  Couldn't the degraded object potentially get fixed when say an OSD starts that happens to have a copy of the require object?

Reply:
Below here we know we are primary and the object does not appear in our local missing list, otherwise it should be considered as an unreadable object and `PrimaryLogPG::do_op` should have prevented it from going this far. If this object is also missing/broken on one or more of our replicas, we'll try to do push (by using our local copy) for the later recovery process, which will end up with another kind of crash since reading it from primary will now return EIO.
 
On a second thought, perhaps we should still add this object in our local missing list and call maybe_kick_recovery instead? As we know we are not clean below here...